src/development/platform-integration/ios/app-extensions.md

```diff
@@ -0,0 +1,213 @@
+options:
+
+- **Key/value:** Use the [`shared_preference_app_group`][]
+  plugin to read or write to `UserDefaults` within the same App Group.
+- **File:** Use the App Group container path from the
+  [`path_provider`][] plugin to [read and write files][].
+- **Database:** Use the App Group container path from
+  the [`path_provider`][] plugin to create a database with the
+  [`sqflite`][] plugin.
+
+### Background updates
+Background tasks provide a means to update your extension through code
+regardless of the status of your app.
+
+To schedule background work from your Flutter app, use the
+[`workmanager`][] plugin.
+
+### Deep linking
+You might want to direct users from an app extension to a
+specific page in your Flutter app.
+To have a URL open a specified route in your app, you can use
+[Deep Linking][].
+
+## Creating app extension UIs with Flutter
+Some app extensions display a user interface.
+For example, iMessage extensions allow users to access your app's
+content directly from the **Messages** app.
+
+<figure class="site-figure {{include.class}}">
+    <div class="site-figure-container">
+        <img src='/assets/images/docs/development/platform-integration/app-extensions/imessage-extension.png'
+        height='300'>
+    </div>
+</figure>
+
+The Flutter command line tool does **not** support 
+building Flutter UI for app extensions. 
+To create the UI for 
+an app extension using Flutter, you must compile 
+a custom engine and embed the `FlutterViewController`
+as described in the following section.
+
+{{site.alert.note}}
+  This requires a custom build of the Flutter engine and
+  is only for advanced users.
+  To learn more, check out [Compiling the engine][].
+{{site.alert.end}}
+
+<!-- TO DO: replace link with a ghost PR -->
+1. Create a custom build of the Flutter engine that removes uses of
+   `sharedApplication` and corrects the path for the bundle.
+   Check out an [example from the community on GitHub][].
+
+1. Open the Flutter app project settings in Xcode to share build
+   configurations. 
+
+   In the **Info** tab,
+   under the **Configurations** expandable group,
+   expand the **Debug**, **Profile**, and **Release** entries.
+   For each, select the same value from the drop-down menu for the
+   extension target as the entry selected for the normal app target. 
+
+    <figure class="site-figure {{include.class}}">
+        <div class="site-figure-container">
+            <img src='/assets/images/docs/development/platform-integration/app-extensions/xcode-configurations.png'
+            height='300'>
+        </div>
+    </figure>
+
+1. If the generated extension code includes a storyboard file, delete it. 
+    In the **Info.plist** file delete the **NSExtensionMainStoryboard** property, 
+    and add add the **NSExtensionPrincipalClass** property with a value that matches the 
+    name of your `ViewController`. For example, in an iMessage extension you would use 
+    `MessageViewController`. 
```

Comment:
**issue:** This reads with difficulty. Suggest breaking it up into a sub-list.
```suggestion
1. (Optional) Replace any storyboard files with an extension class if needed.
    1. In the **Info.plist** file, delete the **NSExtensionMainStoryboard** property.
    1. Add the **NSExtensionPrincipalClass** property.
    1. Set this value for this property to the name of your `ViewController`.
        For example, in an iMessage extension you would use `MessageViewController`. 
```

Reply:
Perfect, thanks!